app/src/main/java/dev/arkbuilders/arkmemo/data/repositories/TextNotesRepoImpl.kt

```diff
@@ -88,13 +91,19 @@ class TextNotesRepoImpl @Inject constructor(): TextNotesRepo {
         persistNoteProperties(resourceId = id, noteTitle = note.content.title)
 
         val resourcePath = root.resolve("$id.$NOTE_EXT")
+        val isDestinationExisting = File(resourcePath.toUri()).exists()
```

Comment:
Do we really have to convert path to Uri and use ```File``` instance?

Reply:
@tuancoltech is it realistic amount of work for this PR to wire index update flow from `arklib-android` into Memo? This should be the right way to establish awareness of paths in the folder.

Reply:
@ShubertMunthali Oh yes, you're right :)
I updated that.

Reply:
@kirillt I haven't got what that `index update flow` actually does for the Memo app.
Do you have an example (e.g, a commit hash) which does the same to other app?

Reply:
@tuancoltech yes, in Navigator: https://github.com/ARK-Builders/ARK-Navigator/blob/main/app/src/main/java/dev/arkbuilders/navigator/presentation/screen/resources/ResourcesPresenter.kt

So the idea is that `arklib` handles all indexing and we can update the index using `updateAll` or `updateOne` which result in updates coming via the flow to the consumer. Ideally, the handling of all resources should be done reactive.

Reply:
It should be considered as refactoring at this moment, the PR serves different purpose. But this async handling of updates would remove necessity to manually check `exists`.

Reply:
@kirillt That's cool. Yes, I took note of this and will look into the indexing part.
Thanks for the explanation!

Reply:
By the way, I'm just curious how the indexing handles the existing file when creating duplicated file? What is its resolution strategy? Rename?

Reply:
Ah, that's tough question at the moment. We need to improve `arklib` regarding duplicates and hash collisions. Basically, in rare cases when several resources map into same _id_, we just take only one resource, ignore the others. All metadata linked to the _id_ points to all the collided resources though. We need to solve this: https://github.com/ARK-Builders/arklib/issues/58

Exact duplicates always have same _id_, but I don't think it's a problem in this case. Because user cares about the content not the location, that's the idea of our project. Tricky situation is when duplicates come from different folders, but this is relevant only to Navigator with its aggregated mode.